src/twisted/web/microdom.py

```diff
@@ -623,7 +622,7 @@ def writexml(self, stream, indent='', addindent='', newl='', strip=0,
             # namespace.  Nothing extra to do here.
             bext(self.tagName)
 
-        j = ''.join
```

Comment:
Same as above comment.  Please feel free to use any old hacky workaround to achieve this; wrapping an underlying `BytesIO` in a `TextIO` wrapper and leaving the rest of the code exactly as you've changed it would be fine.